metrics-operator/api/v1alpha3/analysis_types.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// AnalysisSpec defines the desired state of Analysis
+type AnalysisSpec struct {
+	//Timeframe specifies the range for the corresponding query in the AnalysisValueTemplate
+	Timeframe `json:"timeframe"`
+	// Baggage corresponds to a map of key values pairs that can be used to substitute placeholders in the AnalysisValueTemplate query. The placeholder must be the capitalized version of the key, i.e. for a baggage foo:bar the query could be "query:percentile(95)?scope=tag(my_foo_label:{{.Foo}})".
+	Baggage map[string]string `json:"baggage,omitempty"`
+	// AnalysisDefinition refers to the name of the AnalysisDefinition, a CRD that stores the AnalysisValuesTemplates
+	AnalysDefinition string `json:"analysisDefinition"`
+}
+
+//// AnalysisStatus defines the observed state of Analysis
+//type AnalysisStatus struct {
+//	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+//	// Important: Run "make" to regenerate code after modifying this file
+//	// this will be used later
+//}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// Analysis is the Schema for the analyses API
+type Analysis struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AnalysisSpec `json:"spec,omitempty"`
+	Status string       `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// AnalysisList contains a list of Analysis
+type AnalysisList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Analysis `json:"items"`
+}
+
+type Timeframe struct {
+	// From is the time of start for the query
+	From metav1.Time `json:"from"`
```

Comment:
Should we also specify which time format we expect here? Looking at https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/time.go , RFC3339 should be what we expect here